app/uk/gov/hmrc/apinotificationpull/connectors/ApiNotificationQueueConnector.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apinotificationpull.connectors
+
+import javax.inject.Inject
+
+import uk.gov.hmrc.apinotificationpull.model.Notifications
+import uk.gov.hmrc.apinotificationpull.config.ServiceConfiguration
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ApiNotificationQueueConnector @Inject()(config: ServiceConfiguration, http: HttpClient) {
+
+  private lazy val serviceBaseUrl: String = config.baseUrl("api-notification-queue")
+
+  private val CLIENT_ID_HEADER_NAME = "X-Client-ID"
+
+  private def addHeaders(clientId: String)(implicit hc: HeaderCarrier): HeaderCarrier = {
+    hc.withExtraHeaders(CLIENT_ID_HEADER_NAME -> clientId)
```

Comment:
I don't think you should need to add this in as the x-client-id should be being injected from WSO2 and will be in the header carrier already.

Reply:
You are right.
It is in there already if we make a call to a restricted endpoint.